be/test/olap/rowset/unique_rowset_id_generator_test.cpp

```diff
@@ -0,0 +1,101 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "olap/rowset/unique_rowset_id_generator.h"
+
+#include <gtest/gtest.h>
+#include <iostream>
+
+namespace doris {
+class UniqueRowsetIdGeneratorTest : public testing::Test {
+public:
+    UniqueRowsetIdGeneratorTest() { }
+    virtual ~UniqueRowsetIdGeneratorTest() {
+    }
+};
+
+TEST_F(UniqueRowsetIdGeneratorTest, RowsetIdFormatTest) {
+    int64_t max_id = 1;
+    max_id = max_id << 56;
+    {
+        RowsetId rowset_id;
+        rowset_id.init(123);
+        ASSERT_TRUE(rowset_id.version == 1);
+        ASSERT_TRUE(rowset_id.lo == (123 + max_id));
+        ASSERT_TRUE(rowset_id.mi == 0);
+        ASSERT_TRUE(rowset_id.hi == 0);
+        ASSERT_STREQ("123", rowset_id.to_string().c_str());
+    }
+    {
+        RowsetId rowset_id;
+        rowset_id.init("123");
+        ASSERT_TRUE(rowset_id.version == 1);
+        ASSERT_TRUE(rowset_id.lo == (123 + max_id));
+        ASSERT_TRUE(rowset_id.mi == 0);
+        ASSERT_TRUE(rowset_id.hi == 0);
+        ASSERT_STREQ("123", rowset_id.to_string().c_str());
+    }
+    
+    {
+        RowsetId rowset_id;
+        rowset_id.init("c04f58d989cab2f2efd45faa204491890200000000000003");
+        ASSERT_TRUE(rowset_id.version == 2);
+        ASSERT_TRUE(rowset_id.lo == (3 + max_id));
+        ASSERT_STREQ("c04f58d989cab2f2efd45faa204491890200000000000003", rowset_id.to_string().c_str());
+    }
+}
+
+
+TEST_F(UniqueRowsetIdGeneratorTest, GenerateIdTest) {
```

Comment:
add a case to test BE restart situation.

Reply:
Done